src/Arcus.Observability.Tests.Unit/Correlation/DefaultCorrelationInfoAccessorTests.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Arcus.Observability.Correlation;
+using Xunit;
+
+namespace Arcus.Observability.Tests.Unit.Correlation
+{
+    public class DefaultCorrelationInfoAccessorTests
+    {
+        [Fact]
+        public async Task SetCorrelationInfo_Twice_UsesMostRecentValue()
+        {
+            // Arrange
+            var firstOperationId = $"operation-{Guid.NewGuid()}";
+            var secondOperationId = $"operation-{Guid.NewGuid()}";
+            var transactionId = $"transaction-{Guid.NewGuid()}";
+            await SetCorrelationInfo(firstOperationId, transactionId);
+
+            // Act
+            await SetCorrelationInfo(secondOperationId, transactionId);
+
+            // Assert
+            CorrelationInfo correlationInfo = DefaultCorrelationInfoAccessor.Instance.GetCorrelationInfo();
+            Assert.Equal(secondOperationId, correlationInfo.OperationId);
+            Assert.Equal(transactionId, correlationInfo.TransactionId);
+        }
+
+        private async Task SetCorrelationInfo(string operationId, string transactionId)
```

Comment:
Maybe use a void since nothing here is async?
```suggestion
        private void SetCorrelationInfo(string operationId, string transactionId)
```

Reply:
No, this was explicitly done bc the `AsyncLocal` was using a different value for each async operation. This made sure I could reproduce this.